tests/tokenizers/test_registry.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2024 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Dict, List, Optional
+
+import pytest
+from transformers import BatchEncoding, PreTrainedTokenizer
+
+from llmfoundry.registry import tokenizers
+from llmfoundry.utils import build_tokenizer
+
+
+class DummyTokenizer(PreTrainedTokenizer):
+    """A dummy tokenizer that inherits from ``PreTrainedTokenizer``."""
+
+    def __init__(
+        self,
+        model_name: Optional[str] = 'dummy',
+        **kwargs: Optional[Dict[str, Any]],
+    ):
+        """Dummy constructor that has no real purpose."""
+        super().__init__(
+            model_name=model_name,
+            eos_token='0',
+            pad_token='1',
+            **kwargs,
+        )
+
+    def get_vocab(self) -> Dict[str, int]:
+        return {}
+
+    def tokenize(self, text: str) -> List[str]:
+        return [text]
+
+    def encode(self, text: str) -> List[int]:
+        return [ord(character) for character in text]
+
+    def decode(self, token_ids: List[int]) -> str:
+        return ''.join([chr(token) for token in token_ids])
+
+    def __call__(self, text: str) -> BatchEncoding:
+        raise NotImplementedError()
+
+
+def test_tokenizer_registry():
+    tokenizers.register('dummy', func=DummyTokenizer)
+    tokenizer = build_tokenizer(tokenizer_name='dummy', tokenizer_kwargs={})
+    assert type(tokenizer) == DummyTokenizer
+
+    text = 'Hello World'
```

Comment:
you can delete everything below this for the test (and your dummy class doesn't need to implement anything). the test just needs to test that the object is constructed.

Reply:
is this better? i removed the tokenize, encode, decode, and __call__ methods.